### DISCUSSION TITLE
Lazy load/paginate options in a Combobox

### DISCUSSION THREAD

**Discussion by apcjs** [mantinedev/mantine#5928] [Feature requests]
Is there a way to load options in a Combobox on demand? I have a huge list of items to show in a dropdown. I want to show the first 20 and then when the user scrolls down, I want to retrieve another 20 records and append until all the records are retrieved. How can this be done?

**Comment by rtivital** [comment]
Mantine does not provide this feature out of the box. You need to implement it on your side by creating fetching and scrolling logic for Combobox.Dropdown.

**Reply by apcjs** [reply]
Thanks, I'm trying to do that. But how do I trap the event for user reaching "end of scroll" for a Combobox? I have tried, onScroll, onSeeked, onKeyDown, but none of them seem to trigger on a Combobox.Option

```
<Combobox.Option value={item.val("Name")} key={item.val("Name")}
	onScroll={() => console.log("scroll")}
	onSeeked={() => console.log("seek")}
	onKeyDown={() => console.log("key")}>
		{item.val("Name")}
</Combobox.Option>

<Combobox.Dropdown
	onScroll={() => console.log("scroll")}
	onKeyDown={() => console.log("key")}>
```

**Reply by jamalsoueidan** [reply]
You need to use ref to get the component that hold the scroll and then listen to that :)

**Reply by apcjs** [reply]
Thanks @jamalsoueidan, will try that
